### PR TITLE
add handling for breakout rooms

### LINKED ISSUE #2556: 🖖 Breakout rooms
Meta ticket at https://github.com/nextcloud/spreed/issues/8337

- [ ] TBD

### PR DESCRIPTION
**Pull request by mahibi**
resolve #2556

WIP

Signed-off-by: Marcel Hibbe <dev@mhibbe.de>

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 TODO

- [ ] send raise hand signaling messages
- [ ] avoid crash when opening other chat while being in a call (other PR?). Maybe add something to ChatController like:
  ```
     if (CallActivity.active && roomToken != ApplicationWideCurrentRoomHolder.getInstance().currentRoomToken) {
              router.popCurrentController()
              Toast.makeText(context, "not possible to join other room while calling", Toast.LENGTH_LONG).show()
              return
          }
  ```

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [x] 🔖 Capability is checked or not needed 
- [x] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [x] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/activities/CallActivity.java [hunk 1]
```diff
@@ -390,6 +412,33 @@ public void onCreate(Bundle savedInstanceState) {
...
+        raiseHandViewModel.setData(roomToken, isBreakoutRoom);
+
+        raiseHandViewModel.getViewState().observe(this, viewState -> {
+            if (viewState instanceof RaiseHandViewModel.RaisedHandState) {
+                binding.lowerHandButton.setVisibility(View.VISIBLE);
+            } else if (viewState instanceof RaiseHandViewModel.LoweredHandState) {
+                binding.lowerHandButton.setVisibility(View.GONE);
+            }
+
+            // TODO: build&send raiseHand message (if not possible in RaiseHandViewModel, just do it here..)
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/activities/CallActivity.java:425]: TODO...

File: app/src/main/java/com/nextcloud/talk/webrtc/PeerConnectionWrapper.java [hunk 2]
```diff
@@ -190,6 +190,24 @@ public PeerConnectionWrapper(PeerConnectionFactory peerConnectionFactory,
         }
     }
 
+    public void raiseHand(Boolean raise) {
+        // TODO: build&send raiseHand message (either here or via RaiseHandViewModel)
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply app/src/main/java/com/nextcloud/talk/webrtc/PeerConnectionWrapper.java:194]: Create a follow up ticket for tracking?

File: app/src/main/java/com/nextcloud/talk/raisehand/RequestAssistanceRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
```
Comment (nickvergessen) [inline app/src/main/java/com/nextcloud/talk/raisehand/RequestAssistanceRepositoryImpl.kt:20]:
```suggestion
```

double license

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -628,6 +630,10 @@
     <string name="nc_expire_message_one_hour">1 hour</string>
     <string name="nc_expire_messages_explanation">Chat messages can be expired after a certain time. Note: Files shared in chat will not be deleted for the owner, but will no longer be shared in the conversation.</string>
 
+    <!-- Breakout rooms -->
+    <string name="switch_to_main_room">Switch to main room</string>
```
Comment (nickvergessen) [inline app/src/main/res/values/strings.xml:634]: Not sure about "main room", but can adjust later if we use a different wording in Web/iOS

**Review by nickvergessen — APPROVED** [review]
Looks okay, but didn't test

**Pull request merged**